libbpf-rs/tests/test.rs

```diff
@@ -1191,6 +1193,55 @@ fn test_sudo_object_ringbuf_with_closed_map() {
     });
 }
 
+#[test]
+fn test_sudo_object_user_ringbuf() {
+    #[repr(C)]
+    struct MyStruct {
+        key: u32,
+        value: u32,
+    }
+
+    bump_rlimit_mlock();
+
+    let mut obj = get_test_object("user_ringbuf.bpf.o");
+    let prog = obj
+        .prog_mut("handle__sched_switch")
+        .expect("failed to find program");
+    let _link = prog.attach().expect("failed to attach prog");
+    let urb_map = obj
+        .map("user_ringbuf")
+        .expect("failed to find user ringbuf map");
+    let user_ringbuf = UserRingBuffer::new(urb_map).expect("failed to create user ringbuf");
+    let mut urb_sample = user_ringbuf
+        .reserve::<MyStruct>()
+        .expect("failed to reserve space");
+    let my_struct = urb_sample.as_mut();
+    my_struct.key = 42;
+    my_struct.value = 1337;
+    user_ringbuf
+        .submit(urb_sample)
+        .expect("failed to submit sample");
+
+    // Sleep to ensure the sched_switch event is triggered
+    sleep(Duration::from_millis(10));
+
+    // At this point, the BPF program should have run and consumed the sample in
+    // the user ring buffer, and stored the key/value in the samples map.
+    let samples_map = obj.map("samples").expect("failed to find map");
+    let key: u32 = 42;
+    let value: u32 = 1337;
+    let res = samples_map
+        .lookup(&key.to_ne_bytes(), MapFlags::ANY)
+        .expect("failed to lookup")
+        .expect("failed to find value for key");
+
+    // The value in the samples map should be the same as the value we submitted
+    assert_eq!(res.len(), size_of::<u32>());
+    let mut array = [0; size_of::<u32>()];
+    array.copy_from_slice(&res[..]);
+    assert_eq!(u32::from_ne_bytes(array), value);
+}
+
```

Comment:
Can we also test error cases such as E2BIG?
```rust
#[test]
fn test_sudo_object_user_ringbuf_reservation_too_big() {
    bump_rlimit_mlock();

    let mut obj = get_test_object("user_ringbuf.bpf.o");
    let prog = obj
        .prog_mut("handle__sys_enter_getpid")
        .expect("failed to find program");
    let _link = prog.attach().expect("failed to attach prog");
    let urb_map = obj
        .map("user_ringbuf")
        .expect("failed to find user ringbuf map");
    let user_ringbuf = UserRingBuffer::new(urb_map).expect("failed to create user ringbuf");
    let err = user_ringbuf
        .reserve::<[u8; 1024 * 1024]>()
        .unwrap_err();
    assert!(err.to_string().contains("Requested size is too large"), "{err:#}");
}
```

Reply:
Added test cases for reservation too big and not enough space errors